src/plugins/passwd/CMakeLists.txt

```diff
@@ -1,20 +1,20 @@
+include (CheckFunctionExists)
 include (LibAddMacros)
 
+set (plugin passwd)
+
 if (DEPENDENCY_PHASE)
-	try_compile (TEST_FGETPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasfgetpwent.c")
-	try_compile (TEST_PUTPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasputpwent.c")
+	check_function_exists (fgetpwent TEST_FGETPWENT)
+	check_function_exists (putpwent TEST_PUTPWENT)
+	check_function_exists (getpwent TEST_GETPWENT)
 
-	if (TEST_PUTPWENT)
-		set (PUTPWENT 1)
-	else ()
-		set (PUTPWENT 0)
-	endif ()
 	if (TEST_FGETPWENT)
-		set (FGETPWENT 1)
+		add_definitions (-D_GNU_SOURCE)
```

Comment:
Could you please also shortly describe in the README.md when which function is used. (What the deps are.)

Reply:
Thanks for that, I just noticed putpwent is a terrible choice of function and should probably be replaced with setpwent.

Reply:
* ed28db3 
* 46c545d